### ISSUE TITLE
Stackoverflow error with empty code block

### ISSUE THREAD

**Issue by tristanjuricek** [tristanjuricek/knockoff#17]
I had a basic document started and got a stack overflow error. It looked like this:

```
# Title #
```

Note that the second line has no spaces and the third line has four spaces (start a code block). This made things go kaboom.

**Comment by tristanjuricek** [comment]
Hm, the GitHub markdown didn't work.

**Comment by razie** [comment]
I hit the same issue with a simple sequence of 5 spaces:

```
object Test extends App {
  import com.tristanhunt.knockoff.DefaultDiscounter._
  println(toXHTML(knockoff("""     """)).toString)
}
```

**Comment by tristanjuricek** [comment]
Should be fixed in 0.8.1
